### ISSUE TITLE
Errors trying to upgrade Prickle to Scalajs 0.6.0-RC1

### ISSUE THREAD

**Issue by benhutchison** [scala-js/scala-js#1461]
Background: https://groups.google.com/forum/#!topic/scala-js/NLz_g766hEQ

Code: https://github.com/benhutchison/prickle/commit/b9c4ef6d9b686c9

Current error: Some keys were defined with the same name but different types: 'jsDependencies' (scala.collection.Seq[scala.scalajs.sbtplugin.AbstractJSDep], scala.collection.Seq[org.scalajs.sbtplugin.AbstractJSDep])

**Comment by jducoeur** [comment]
Don't know that particular code, but that smells like a type alias gone awry -- when I hit errors like that, it is usually because I have both an actual type and an alias in scope, and the compiler isn't recognizing that they're actually the same type.  Are org.scalajs.sbtplugin and scala.scalajs.sbtplugin actually the same type or something?

**Comment by sjrd** [comment]
You must remove your dependency on the uTest plugin:
```scala
addSbtPlugin("com.lihaoyi" % "utest-js-plugin" % "0.2.4")
```
(or upgrade it to a version compatible with 0.6.x)
This one pulls in the Scala.js sbt plugin of 0.5.x in the classpath of your build, which obviously causes all sorts of problems.

**Comment by benhutchison** [comment]
Ok, helpful. Thanks! 

I switched to `addSbtPlugin("com.lihaoyi" % "utest-js-plugin" % "0.2.5-RC1")` and the project loaded OK. Also, I found that since then, `%%%` resolved correctly even without the import @japgolly had suggested `import org.scalajs.sbtplugin.ScalaJSPlugin.autoImport.toScalaJSGroupID`.

Next problem: I tried to `publishLocal` a 0.6.0 version of Prickle's dependency Microjson [0.6.0 branch here](https://github.com/benhutchison/MicroJson/commit/d7c89dc1e7bdda1). However, the generated artifact has no classfiles or JS inside; ie once I unzipped jar:
```
[ben_hutchison@VIC009228M MicroJson Scalajs0.6.0]$ cd js/target/scala-2.11/microjson_sjs0.6.0-RC1_2.11-1.1-RC1
[ben_hutchison@VIC009228M microjson_sjs0.6.0-RC1_2.11-1.1-RC1 Scalajs0.6.0]$ ls
JS_DEPENDENCIES	META-INF
``` 

Any ideas? A missing build setting? A required command not run?

**Comment by sjrd** [comment]
Well, with your branch, I don't even have anything to *compile*.
```
> show js/sources
[info] ArrayBuffer()
[success] Total time: 0 s, completed 17-janv.-2015 14:51:52
> show jvm/sources
[info] ArrayBuffer()
[success] Total time: 0 s, completed 17-janv.-2015 14:51:55
```
No wonder the published artifact contains nothing.

**Comment by benhutchison** [comment]
Oops! Symlinks to `shared/` are not getting transferred through git. 
Once fixed, could build both libraries with 0.6.0 successfully. Thanks for your help.
